main.go

```diff
@@ -143,56 +147,117 @@ func main() {
 		ReconcileTimeout: reconcileTimeout,
 		WatchFilterValue: watchFilterValue,
 	}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpClusterConcurrency}); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "GCPCluster")
-		os.Exit(1)
+		return fmt.Errorf("setting up GCPCluster controller: %w", err)
 	}
 
-	if err = (&infrav1beta1.GCPCluster{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "GCPCluster")
-		os.Exit(1)
+	if feature.Gates.Enabled(feature.GKE) {
+		if err := (&expcontrollers.GCPManagedClusterReconciler{
+			Client:           mgr.GetClient(),
+			ReconcileTimeout: reconcileTimeout,
+			WatchFilterValue: watchFilterValue,
+		}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpClusterConcurrency}); err != nil {
+			return fmt.Errorf("setting up GCPManagedCluster controller: %w", err)
+		}
+		if err := (&expcontrollers.GCPManagedControlPlaneReconciler{
+			Client:           mgr.GetClient(),
+			ReconcileTimeout: reconcileTimeout,
+			WatchFilterValue: watchFilterValue,
+		}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpClusterConcurrency}); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "GCPManagedControlPlane")
+			os.Exit(1)
+		}
+		if err := (&expcontrollers.GCPManagedMachinePoolReconciler{
+			Client:           mgr.GetClient(),
+			ReconcileTimeout: reconcileTimeout,
+			WatchFilterValue: watchFilterValue,
+		}).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: gcpClusterConcurrency}); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "GCPManagedMachinePool")
+			os.Exit(1)
+		}
 	}
-	if err = (&infrav1beta1.GCPClusterTemplate{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "GCPClusterTemplate")
-		os.Exit(1)
+
+	return nil
+}
+
+func setupWebhooks(mgr ctrl.Manager) error {
+	if err := (&infrav1beta1.GCPCluster{}).SetupWebhookWithManager(mgr); err != nil {
+		return fmt.Errorf("setting up GCPCluster webhook: %w", err)
 	}
-	if err = (&infrav1beta1.GCPMachine{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "GCPMachine")
-		os.Exit(1)
+	if err := (&infrav1beta1.GCPClusterTemplate{}).SetupWebhookWithManager(mgr); err != nil {
+		return fmt.Errorf("setting up GCPClusterTemplate webhook: %w", err)
+	}
+	if err := (&infrav1beta1.GCPMachine{}).SetupWebhookWithManager(mgr); err != nil {
+		return fmt.Errorf("setting up GCPMachine webhook: %w", err)
+	}
+	if err := (&infrav1beta1.GCPMachineTemplate{}).SetupWebhookWithManager(mgr); err != nil {
+		return fmt.Errorf("setting up GCPMachineTemplate webhook: %w", err)
+	}
+
+	if feature.Gates.Enabled(feature.GKE) {
+		if err := (&infrav1exp.GCPManagedCluster{}).SetupWebhookWithManager(mgr); err != nil {
```

Comment:
Same comment as above with enabling the GKE controllers in the next PR

Reply:
reverted